neqo-transport/src/recovery.rs

```diff
@@ -968,7 +968,9 @@ impl LossRecovery {
         }
         self.stats.borrow_mut().lost += lost_packets.len();
 
-        self.maybe_fire_pto(primary_path.borrow().rtt(), now, &mut lost_packets);
+        if primary_path.borrow().is_valid() {
+            self.maybe_fire_pto(primary_path.borrow().rtt(), now, &mut lost_packets);
+        }
```

Comment:
Perhaps the right thing to do is pass the `PathRef` into `maybe_fire_pto`, so that the "maybe" can do the extra work (and so you can borrow just once).